rust/flatbuffers/src/endian_scalar.rs

```diff
@@ -28,22 +28,27 @@ use core::mem::size_of;
 /// "too much". For example, num-traits provides i128 support, but that is an
 /// invalid FlatBuffers type.
 pub trait EndianScalar: Sized + PartialEq + Copy + Clone {
-    fn to_little_endian(self) -> Self;
-    fn from_little_endian(self) -> Self;
+    type Scalar;
```

Comment:
This change is necessary to support big endian bitflags. In particular the previous logic would use a combination of https://docs.rs/bitflags/latest/bitflags/example_generated/struct.Flags.html#method.bits and https://docs.rs/bitflags/latest/bitflags/example_generated/struct.Flags.html#method.from_bits_unchecked to reverse the endianness. This is technically ill-formed, as it would create bitflags with out of bounds values.

Allowing EndianScalar to return a different type solves this problem, and also simplifies the implementation for floats